app/src/main/java/org/mozilla/fenix/search/toolbar/ToolbarView.kt

```diff
@@ -152,7 +153,7 @@ class ToolbarView(
 
         val searchEngine = searchState.searchEngineSource.searchEngine
 
-        if (searchEngine != null) {
+        if (!context.settings().showUnifiedSearchFeature && searchEngine != null) {
```

Comment:
This will hide the search icon when you're editing in the toolbar if the FeatureFlag is turned on.